dotnet/WPF/VisualLayerIntegration/VisualLayerIntegration/BarGraphHostControl.xaml.cs

```diff
@@ -159,19 +160,19 @@ public void UpdateGraph()
                 var yAxisTitle = "# Shares of Stock";
 
                 // If graph already exists update values. Otherwise, create new graph.
```

Comment:
Why not create the graph in the constructor so you don't have to have all these checks for _currentGraph?

Reply:
requires a non-null compositor, which did not work in the constructor

Reply:
See my commments in CompositionHost - it should allow you to get the Compositor at any time, and the CompositionHost is passed to the constructor of this class, so all of this can move into the constructor, more things can be readonly, and the checks for _currentGraph can be eliminated.

---
In reply to: [274674429](https://github.com/Microsoft/Windows.UI.Composition-Win32-Samples/pull/1#discussion_r274674429) [](ancestors = 274674429)